tests/test_cases/test_cocotb/test_timing_triggers.py

```diff
@@ -248,3 +248,9 @@ async def test_singleton_isinstance(dut):
     assert isinstance(NextTimeStep(), NextTimeStep)
     assert isinstance(ReadOnly(), ReadOnly)
     assert isinstance(ReadWrite(), ReadWrite)
+
+
+@cocotb.test(expect_fail=True, expect_error=TriggerException)
+def test_neg_timer(dut):
+    """Test intentionally failing by requesting negative timer value"""
+    yield Timer(-42)
```

Comment:
Better to verify the exception is thrown rather than that the test fails:
```suggestion
@cocotb.test()
async def test_neg_timer(dut):
    """Test intentionally failing by requesting negative timer value"""
    with assert_raises(TriggerException):
        Timer(-42)  # no need to even `await`, constructing it is an error
```

Reply:
Doesn't expect_error=TriggerException do exactly this?

Reply:
Or I should say: then probably some other tests in that suite should be rewritten as well.

Reply:
We intending to use that feature only for exempting tests that fail on certain simulators.

>  some other tests in that suite should be rewritten as well.

Absolutely they do, but there are a freaking lot of them that need to be rewritten... That is outside the scope of this PR, so don't worry about them for right now.

Reply:
Getting your point for the expect_fail parameter, but how would that apply to expect_error.

From what I read its feature set basically exactly mirrors assert_raises and in my view it is much more elegant that way.

Because now we introduce 2 flavors of intentionally failing tests, which looks a bit inconsistent:

a) some tests tell you right away in the decorator they will fail and do so intentionally
b) some other tests look as if they did something 'useful', but truly they also fail, but you can only tell after looking at the code

Bit complicated to explain, you get my point?

Reply:
Also: I think cocotb produces a very nice error output by sying: 'failing as expected, Result was TriggerException'

In the other instance the test would just pass, almost implying you could use negative timer values.

Reply:
There are two simple answers for why `assert_raises` is a better method:

* It's the python way - it's how `pytest` and `unittest` do things
* It lets you test multiple errors in a single test
  ```python
      with assert_raises(TriggerException):
          Timer(-42)
      with assert_raises(TriggerException):
          Timer(-1)
  ```

Reply:
> some tests tell you right away in the decorator they will fail and do so intentionally

Following the example of `pytest` and `unitest`, the only reasons tests should fail intentionally are:

* You know the test doesn't yet pass, but you'd like it to pass in future
* The test started breaking and you want to put off fixing it till later.

Semantically, "the test failed" and "my test that an exception was thrown succeeded" are very different things.

Reply:
Ok, so you are saying you basically wish to move to method b) completely, unless there is some technical reason to let a test fail. Which makes complete sense to me, because then it would be consistent.